### PR TITLE
Subscription

### PR DESCRIPTION
**Pull request by doodko**

### CODE REVIEW

File: migrations/versions/1add42f9be06_add_period_model.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('period',
+    sa.Column('zone', sa.Integer(), nullable=False),
+    sa.Column('is_online', sa.Boolean(), nullable=False),
+    sa.Column('start', sa.DateTime(), nullable=False),
+    sa.Column('end', sa.DateTime(), nullable=False),
+    sa.Column('duration', sa.Integer(), nullable=False),
```
Comment (vitalii-dud) [inline migrations/versions/1add42f9be06_add_period_model.py:26]: Does it make sense to store duration if you already storing `start` and `end`?

File: migrations/versions/57e3e1d9716a_initial_migration.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('zone',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('name', sa.String(), nullable=False),
+    sa.Column('is_online', sa.Boolean(), nullable=False),
```
Comment (vitalii-dud) [inline migrations/versions/57e3e1d9716a_initial_migration.py:24]: Rule of thumb — better to skip those prefixes like `has_`, `is_` or similar. `online: true|false` would give you same understanding but better readability.

File: migrations/versions/57e3e1d9716a_initial_migration.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    sa.Column('updated_at', sa.DateTime(), nullable=False),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table('host',
+    sa.Column('address', sa.String(), nullable=False),
+    sa.Column('zone', sa.Integer(), nullable=False),
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('name', sa.String(), nullable=False),
+    sa.Column('is_online', sa.Boolean(), nullable=False),
```
Comment (vitalii-dud) [inline migrations/versions/57e3e1d9716a_initial_migration.py:33]: Same here.

File: migrations/versions/723c650ca61c_add_user_and_subscription_models.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('user',
+    sa.Column('name', sa.String(), nullable=False),
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table('subscription',
+    sa.Column('user_id', sa.Integer(), nullable=False),
+    sa.Column('zone_id', sa.Integer(), nullable=False),
```
Comment (vitalii-dud) [inline migrations/versions/723c650ca61c_add_user_and_subscription_models.py:28]: Can we add uniq constraint so there would be no doubled subscriptions for same user — same zone?

File: ping_app/ping_service.py [hunk 5]
```diff
@@ -83,6 +83,12 @@ async def get_current_zones_status() -> str:
         zone_statuses = [notifier.get_current_state(zone) for zone in zones]
         return '\n'.join(zone_statuses)
 
+    @staticmethod
+    async def get_current_zones_status_short() -> str:
+        zones = await host_crud_service.get_all_zones()
+        zone_statuses = [notifier.get_current_state_short(zone) for zone in zones]
```
Comment (vitalii-dud) [inline ping_app/ping_service.py:89]: I'm not sure but would this work?
```suggestion
        zone_statuses = [notifier.get_current_state_short(zone) for zone in await host_crud_service.get_all_zones()]
```

**Review by vitalii-dud — COMMENTED** [review]
Drop some comments, but overall it feels like responses are unstructured and a bit messy, so it would be great if you can have single entrypoint for zone notification and then use it to get short or long response.

File: bot/handlers/faq.py [hunk 6]
```diff
@@ -30,13 +30,17 @@ async def say_index(message: Message):
...
 
 
-@router.message(F.text.lower().regexp(r".*(\bсвітло\b.*\bє\b).*|.*(\bє\b.*\bсвітло\b).*\?"))
+@router.message(F.text.lower().regexp(r".*((\bсвітло\b.*\bє\b).*|.*(\bє\b.*\bсвітло\b)).*\?"))
 async def say_current_status(message: Message):
-    text = await ps.get_current_zones_status()
+    if message.chat.type in ('group', 'supergroup'):
+        text = await ps.get_current_zones_status_short()
+    elif message.chat.type == 'private':
+        text = await ps.get_current_zones_status()
```
[the remark this answers is not in the document]
Comment (doodko) [reply bot/handlers/faq.py:38]: fully agree!

File: bot/handlers/faq.py [hunk 7]
```diff
@@ -30,13 +30,17 @@ async def say_index(message: Message):
...
+
     await message.reply(text=text)
 
 
-@router.message(F.text.lower().regexp(r".*(\bсвет\b.*\bесть\b).*|.*(\bесть\b.*\bсвет\b).*\?"))
+@router.message(F.text.lower().regexp(r".*((\bсвет\b.*\bесть\b).*|.*(\bесть\b.*\bсвет\b)).*\?"))
```
[the remark this answers is not in the document]
Comment (doodko) [reply bot/handlers/faq.py:43]: fo sure, done

File: migrations/README [hunk 8]
```diff
@@ -0,0 +1 @@
+Generic single-database configuration.
```
[the remark this answers is not in the document]
Comment (doodko) [reply migrations/README:1]: this is auto generated file

File: migrations/versions/1add42f9be06_add_period_model.py (hunk 1 above)
Comment (doodko) [reply migrations/versions/1add42f9be06_add_period_model.py:26]: probably not, more time is needed for a better solution

File: migrations/versions/57e3e1d9716a_initial_migration.py (hunk 2 above)
Comment (doodko) [reply migrations/versions/57e3e1d9716a_initial_migration.py:24]: thnx!

File: migrations/versions/723c650ca61c_add_user_and_subscription_models.py (hunk 4 above)
Comment (doodko) [reply migrations/versions/723c650ca61c_add_user_and_subscription_models.py:28]: was going to check the uniqueness of the subscription while adding a new one, not sure how to constraint it by db

File: ping_app/ping_service.py (hunk 5 above)
Comment (doodko) [reply ping_app/ping_service.py:86]: this will work, but worse readability imho

**Pull request merged**